src/pykiso/lib/auxiliaries/udsaux/common/odx_parser.py

```diff
@@ -0,0 +1,290 @@
+##########################################################################
+# Copyright (c) 2010-2022 Robert Bosch GmbH
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0
+##########################################################################
+
+"""
+ODX parser
+**********
+
+:module: odx_parser
+
+:synopsis: A basic parser for Open Diagnostics eXchange (ODX) format.
+
+.. currentmodule:: odx_parser
+
+"""
+
+import xml.etree.ElementTree as ET
+from collections import defaultdict
+from pathlib import Path
+from typing import Any, Dict, Union
+
+from uds_utils import SERVICE_ID_TO_NAME
+
+
+class OdxParser:
+    """A parser for ODX files that extracts the request and positive response
+    information and stored it into the `service_configuration` attribute.
+    """
+
+    def __init__(
+        self, odx_file_path: Union[str, Path], **services: Dict[int, str]
+    ) -> None:
+        self.odx_file = odx_file_path
+        self.services = {**SERVICE_ID_TO_NAME, **services}
+        self.service_configuration = defaultdict(dict)
+        self._xml_elements = {}
+
+    def parse(self) -> Dict[int, Dict[str, Any]]:
+        """Parse an ODX file to extract the request and positive response information
+        for each service.
+
+        :return: the ODX file parsed as a dictionary.
+        """
+        root = ET.parse(self.odx_file)
+
+        self._xml_elements = {
+            child.attrib["ID"]: child
+            for child in root.iter()
+            if child.attrib.get("ID") is not None
+        }
+
+        for service_desc in self._xml_elements.values():
```

Comment:
Can you please comment the steps better? What is SDGS for, etc...

Reply:
As discussed I removed the OdxParser for the moment and now throw out a warning if the auxiliary is configured with an ODX file